iroh-net/src/key.rs

```diff
@@ -38,7 +58,7 @@ impl Serialize for PublicKey {
     where
         S: serde::Serializer,
     {
-        self.public.serialize(serializer)
+        self.public().serialize(serializer)
```

Comment:
Why can't this be self.0.serialize?

Reply:
you need to use `serialize_bytes`, see https://github.com/dalek-cryptography/ed25519-dalek/blob/main/src/verifying.rs#L630